### PR TITLE
Game Rating Feature

### PR DESCRIPTION
**Pull request by ctruong1**
Hi everyone. I'm kinda lost on how to finish up #79. For some reason I just cant figure out how to get the form data to summit to the database/game_user table. I've tried doing this a few different ways but still cant get it to work. Helps pls :( any suggestions on how I should do this??

### CODE REVIEW

**Comment by dshea18** [comment]
Hey Colin @ctruong1 , I'm taking a look at this now to help out.  FYI, Sarah (@sarahdactylrex ) and Andrew (@oldskoolfan), I'm going to create a branch to build out some additional code. If anyone else has started on this let me know.

Thanks,
Danny

**Comment by oldskoolfan** [comment]
There's some sections of the Eloquent documentation worth looking at:
https://laravel.com/docs/8.x/eloquent-relationships#retrieving-intermediate-table-columns
https://laravel.com/docs/8.x/eloquent-relationships#customizing-the-pivot-attribute-name
https://laravel.com/docs/8.x/eloquent-relationships#defining-custom-intermediate-table-models

Let me know if I can help with this, curious to see where you get @dshea18

**Review by dshea18 — CHANGES_REQUESTED** [review]
Hey Colin,

I just sent over a pull request for additional things that would be needed to get this feature to work. Take a look and let me know what you think and see if it helps out.

V/R,
Danny

**Review by sarahdactylrex — APPROVED** [review]
Danny, dude, this looks amazing! I barely got a minute to look over the initial commit and it looks like you've solved it! Cool beans!

**Review by sarahdactylrex — DISMISSED** [review]

**Review by ctruong1 — COMMENTED** [review]
Everything looks great!!! Thanks again for your help @dshea18! :)

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -24,4 +31,29 @@ public function userInfo()
...
+    public function ViewGameReview(){
+        return view('YourReview');
+    }
+
+
+    public function SubmitGameRating(UserGameReviewPost $request){
+        // $this->authorize('create', Reviews::class);
+        $validatedInput = $request->validated();
+        $user_id = Auth::user()->id;
+        $userReview = $this->userService->submitGameReview($validatedInput['game_id'],$user_id,$validatedInput['game_rating'],$validatedInput['text_review']);
```
Comment (oldskoolfan) [inline app/Http/Controllers/UserController.php:53]: can we wrap this so the line isn't so long? makes for better readability

File: app/Http/Requests/UserGameReviewPost.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class UserGameReviewPost extends FormRequest
```
Comment (oldskoolfan) [inline app/Http/Requests/UserGameReviewPost.php:7]: Nice!

File: app/Repositories/GameRepository.php [hunk 3]
```diff
@@ -28,5 +28,7 @@ public function getGameById($id)
     {
         return DB::table('games')->find($id);
     }
+
+    
     
 }
```
Comment (oldskoolfan) [inline app/Repositories/GameRepository.php:34]: needs newline at EOF

File: app/Models/Reviews.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Models;
+
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Reviews extends Model
```
Comment (oldskoolfan) [inline app/Models/Reviews.php:9]: I think the many-to-many linking tables need to extend the pivot class based on what I saw here: https://laravel.com/docs/8.x/eloquent-relationships#defining-custom-intermediate-table-models

File: app/Repositories/UserRepository.php [hunk 5]
```diff
@@ -10,4 +13,16 @@ public function getWishlistByUserId(): array
...
+        DB::insert('insert into game_user (game_id, user_id, game_rating, text_review) values (?,?,?,?)', [$game_id, $user_id,$game_rating,$text_review]);
+        $dbValue = DB::table('game_user')->where('game_id', $game_id)->first();
+        $userReview = new Reviews();
+        $userReview->game_id = $dbValue->game_id;
+        $userReview->user_id = $dbValue->user_id;
+        $userReview->game_rating = $dbValue->game_rating;
+        $userReview->text_review = $dbValue->text_review;
+        return $userReview;
+    }
 }
```
Comment (oldskoolfan) [inline app/Repositories/UserRepository.php:28]: needs newline at EOF

File: app/Services/GameService.php [hunk 6]
```diff
@@ -28,5 +28,7 @@ public function searchGamesByTitle (Request $title)
     {    
         return ($this->gameRepository->searchGamesByTitle($title));
     }
+
+    
     
 }
```
Comment (oldskoolfan) [inline app/Services/GameService.php:34]: needs newline at EOF

File: app/Services/UserService.php [hunk 7]
```diff
@@ -15,4 +16,12 @@ public function getWishlistByUserId(): array
...
     }
+
+    public function submitGameReview(int $game_id, int $user_id, float $game_rating, string $text_review):Reviews
+    {
+        $submittedReview = $this->userRepository->InsertGameReview($game_id, $user_id, $game_rating, $text_review);
+        $returnedReview =  new Reviews();
+        $returnedReview = $submittedReview;
+        return ($returnedReview);
+    }
 }
```
Comment (oldskoolfan) [inline app/Services/UserService.php:27]: needs newline at EOF

File: app/Services/UserService.php [hunk 8]
```diff
@@ -15,4 +16,12 @@ public function getWishlistByUserId(): array
     {
         return $this->userRepository->getWishlistByUserId();
     }
+
+    public function submitGameReview(int $game_id, int $user_id, float $game_rating, string $text_review):Reviews
+    {
+        $submittedReview = $this->userRepository->InsertGameReview($game_id, $user_id, $game_rating, $text_review);
+        $returnedReview =  new Reviews();
+        $returnedReview = $submittedReview;
```
Comment (oldskoolfan) [inline app/Services/UserService.php:24]: why not just `return $this->userRepository->InsertGameReview($game_id, $user_id, $game_rating, $text_review);`

File: resources/views/gameView.blade.php [hunk 9]
```diff
@@ -14,18 +14,39 @@
...
-                    </table>  
-                    </table> 
+                    </table>   
                 </div>
             </div>
         </div>
     </div>
+    
+<!-- 
+    <div class="max-w-7xl mx-auto sm:px-6 lg:px-8">
```
Comment (oldskoolfan) [inline resources/views/gameView.blade.php:34]: remove commented code

File: routes/web.php [hunk 10]
```diff
@@ -31,4 +32,17 @@
 
 Route::get('/games', [GameController::class, 'listGames'])->name('games');
 
+// Route::post('/gameView',function(){
```
Comment (oldskoolfan) [inline routes/web.php:35]: remove commented code

**Review by oldskoolfan — CHANGES_REQUESTED** [review]
just some minor suggestions, thank you for helping out with this @dshea18 !

**Comment by ctruong1** [comment]
Hi everyone, I made some new commits with a few updates hope I didn't mess anything up

**Review by oldskoolfan — DISMISSED** [review]
looks good!

**Review by sarahdactylrex — APPROVED** [review]
Excellent work!

**Review by dshea18 — APPROVED** [review]

**Pull request merged**